From 5235c025fe06086df18a6b258bfd979aa3d42527 Mon Sep 17 00:00:00 2001
From: George Kiagiadakis <george.kiagiadakis@collabora.com>
Date: Sun, 2 Jun 2024 15:17:06 +0300
Subject: [PATCH] conf: skip empty configuration files to avoid crashing

g_mapped_file_get_contents() returns NULL if the file is empty

Fixes: #671
---
 lib/wp/conf.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/lib/wp/conf.c b/lib/wp/conf.c
index 6f66e216..0af23960 100644
--- a/lib/wp/conf.c
+++ b/lib/wp/conf.c
@@ -216,6 +216,11 @@ open_and_load_sections (WpConf * self, const gchar *path, GError ** error)
   if (!file)
     return FALSE;
 
+  if (!g_mapped_file_get_contents (file) || g_mapped_file_get_length (file) == 0) {
+    wp_notice_object (self, "Ignoring empty configuration file at '%s'", path);
+    return TRUE;
+  }
+
   /* test if the file is a relic from 0.4 */
   if (detect_old_conf_format (self, file)) {
     g_set_error (error, WP_DOMAIN_LIBRARY, WP_LIBRARY_ERROR_INVALID_ARGUMENT,
-- 
GitLab

